airflow-core/src/airflow/models/serialized_dag.py

```diff
@@ -401,14 +401,16 @@ def write_dag(
         # If Yes, does nothing
         # If No or the DAG does not exists, updates / writes Serialized DAG to DB
         if min_update_interval is not None:
-            if session.scalar(
+            do_nothing = session.scalar(
```

Comment:
Probably better to name this variable something like `has_conflicts`. It describes what this check actually does a bit better, and the _do nothing_ nature is still implied by the immediate `return False` line.